packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -708,10 +708,21 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
         projectDir: globals.fs.currentDirectory,
         buildMode: buildMode,
       );
+
+      final File cachedFlavorFile = globals.fs.file(
+        globals.fs.path.join('build', 'test_cache', 'flavor.txt'),
+      );
+      if (cachedFlavorFile.existsSync()) {
+        await cachedFlavorFile.delete();
+      }
+      if (flavor != null) {
+        await cachedFlavorFile.create(recursive: true);
+        await cachedFlavorFile.writeAsString(flavor);
```

Comment:
For these I would probably just use the sync variants, since this file is going to be super small (hopefully).

Reply:
Addressed in [use sync file ops](https://github.com/flutter/flutter/pull/150461/commits/a3273e0f8ef8495d93f9a31b4a156471e87769be)